src/status_im/contexts/wallet/swap/events.cljs

```diff
@@ -35,34 +36,38 @@
               (assoc-in [:wallet :ui :swap :network] network')
               (assoc-in [:wallet :ui :swap :launch-screen] view-id)
               (assoc-in [:wallet :ui :swap :start-point] start-point))
-      :fx (if network'
-            [[:dispatch [:wallet/switch-current-viewing-account (:address account)]]
-             [:dispatch
-              (if open-new-screen?
-                [:open-modal :screen/wallet.setup-swap]
-                [:navigate-to-within-stack
-                 [:screen/wallet.setup-swap :screen/wallet.swap-select-asset-to-pay]])]
-             [:dispatch
-              [:centralized-metrics/track :metric/swap-start
-               {:network       (:chain-id network)
-                :pay_token     (:symbol asset-to-pay)
-                :receive_token (:symbol asset-to-receive)
-                :start_point   start-point
-                :launch_screen view-id}]]
-             [:dispatch [:wallet.swap/set-default-slippage]]]
-            [[:dispatch
-              [:show-bottom-sheet
-               {:content (fn []
-                           [network-selection/view
-                            {:token-symbol      (:symbol asset-to-pay)
-                             :on-select-network (fn [network]
-                                                  (rf/dispatch [:hide-bottom-sheet])
-                                                  (rf/dispatch
-                                                   [:wallet.swap/start
-                                                    {:asset-to-pay     asset-to-pay
-                                                     :asset-to-receive asset-to-receive
-                                                     :network          network
-                                                     :open-new-screen? open-new-screen?}]))}])}]]])})))
+      :fx (if (and multi-account? (= view-id :wallet-stack) (not from-account))
+            [[:dispatch [:open-modal :screen/wallet.swap-from]]]
```

Comment:
Quick question: From reading the code, i can understand that this opens the account selection screen (from) screen if the user has more than one account (any type). This is different in send/bridge flow. Only if the asset is available on multiple accounts, that's when we open the from screen with list of accounts that has that asset.

cc @briansztamfater

Reply:
yes it's different according to QA's doc https://docs.google.com/spreadsheets/d/1fOg99N5f9Q0zfJQcIqllPU1RK3Y93Uy_CumzlXV1Y9M/edit?gid=0#gid=0

Reply:
according to design we list all accounts but show account with zero balance with disabled state

Reply:
I see, but seems that the test cases differs from the designs. We should double check that cc @pavloburykh @churik 

![](https://github.com/user-attachments/assets/59569e01-f3a5-4a65-ae8b-4917d8914f1a)

Reply:
Oh, then we are doing it wrong in Send/Bridge flow i guess.

Reply:
yeah maybe just condition here has a wrong text, otherwise we have account with zero balance in design.

Reply:
In the designs three accounts are shown, which two of them have balance, and that's why the third one is disabled. But (from my understanding) if the user has two accounts and only balance for the selected asset in only one of them, the account selection page should not be shown.

Reply:
I guess better to check linked issue description,  https://github.com/status-im/status-mobile/issues/21557#issue-2633022911

also list of [list of test cases](https://docs.google.com/spreadsheets/d/1fOg99N5f9Q0zfJQcIqllPU1RK3Y93Uy_CumzlXV1Y9M/edit?gid=0#gid=0)

I implemented it accordingly

Reply:
Implementation looks correct based in the test cases document, I am just wondering if test cases are correctly set, as design implies that the From page should be shown only if asset exist in multiple accounts, but on the document there are some test cases that expect the From page to be always shown if the user has more than one account, no matter if the asset exist on more than one account. I think we should double check test cases.

Reply:
@briansztamfater Thanks for your comments! I’ll review these cases one more time.